ddl/cluster.go

```diff
@@ -37,12 +37,15 @@ func (w *worker) onFlashbackCluster(d *ddlCtx, t *meta.Meta, job *model.Job) (ve
 			if j.ID == job.ID {
 				continue
 			}
+			// has ddl job in queue, return error
 			if !j.IsFinished() {
 				return true, errors.Errorf("Can't flashback cluster, other ddl jobs in ddl queue")
 			}
+			// job is finished in time range [flashbackTS, now], not support flashback now
 			if j.Type != model.ActionFlashbackCluster && j.BinlogInfo != nil && j.BinlogInfo.FinishedTS >= flashbackTS {
 				return true, errors.Errorf("Can't flashback cluster because of ddl history")
 			}
+			// all jobs are done before flashbackTS, stop the iterator
 			if j.BinlogInfo != nil && j.BinlogInfo.FinishedTS < flashbackTS {
```

Comment:
Are jobs in DDL queue sorted by `finishedTS`? Early stop of iterator has such an assumption, right?

Reply:
Yes. I think it's a good way to check ddl history. Maybe use schemaVersion is better? What do you think?

Reply:
> Are jobs in DDL queue sorted by finishedTS

I'm not sure about this.

> Maybe check schemaVersion is better?

I sounds a better idea to me: we could get the snapshot infoschema [`GetSnapshotInfoschema()`](https://github.com/pingcap/tidb/blob/272ea030f0dd28a3540f43589947b09c428575c3/domain/domain.go#L348) and simply check if the schema version has been changed between flashbackTS and now. In this way we don't need to iterate the job history.

Reply:
Done, PTAL @xhebox @bb7133